custom/ewsghana/utils.py

```diff
@@ -58,7 +58,11 @@ def make_url(report_class, domain, string_params, args):
 
 
 def calculate_last_period(enddate):
-    last_th = enddate - timedelta(days=enddate.weekday()) + timedelta(days=3, weeks=-1)
```

Comment:
it's pretty hard to understand what this function is supposed to do - maybe you could add a comment or a doc string?